content/2017.06.kinto-elasticsearch.rst

```diff
@@ -0,0 +1,249 @@
+
+We load it into Kinto using `kinto-http.py <https://github.com/Kinto/kinto-http.py>`_:
+
+.. code-block :: python
+
+    import kinto_http
+
+    server = "http://localhost:8888/v1"
+    bucket = "restaurants"
+    collection = "pizzerias"
+    auth = ("user", "pass")
+
+    # Create the destination bucket.
+    client = kinto_http.Client(server_url=server, auth=auth)
+    client.create_bucket(id=bucket, if_not_exists=True)
+
+We define the index and permissions. Each pizzeria will have a name and a location (long, lat).
+
+.. code-block :: python
+
+    # Define the ElasticSearch mapping in the collection metadata.
+    collection_metadata = {
+        "index:schema": {
+            "properties": {
+                "name": {
+                    "type": "text"
+                },
+                "location": {
+                    "type": "geo_point"
+                }
+            }
+        }
+    }
+    # Let anonymous users read the records (online map demo).
+    collection_permissions = {
+        "read": ["system.Everyone"]
+    }
+    client.create_collection(id=collection, bucket=bucket, data=collection_metadata,
+                             permissions=collection_permissions, if_not_exists=True)
+
+Create a record for each «feature» in the GeoJSON file, using batched requests (one HTTP request includes many creations):
+
+.. code-block :: python
+
+    export = json.load(open("export.geojson"))
+
+    with client.batch() as batch:
+        for pizzeria in export["features"]:
+            record = {
+                "name": pizzeria["properties"].get("name"),
+                "location": pizzeria["geometry"]["coordinates"],
+            }
+            batch.create_record(data=record, bucket=bucket, collection=collection)
+
+
+Web mapping
+-----------
+
+Now that the data is loaded in Kinto, indexed in ElasticSearch, and publicly available, we'll render a map.
+
+Leaflet.js is a nice piece of software that makes webmapping really pleasant.
+
+.. code-block :: javascript
+
+    // A map with OpenStreetMap background.
+    const map = L.map('map');
+    L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
+        attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+    }).addTo(map);
+
+Using `kinto-http.js <https://github.com/Kinto/kinto-http.js/>`_, it is also quite intuitive to retrieve the whole list of records:
+
+.. code-block :: javascript
+
+    // Fetch all records from Kinto to populate the map layer.
+    // (Note: this also could have been done using ES.)
+    const kinto = new KintoClient(server);
+
+    kinto.bucket(bucket).collection(collection).listRecords()
+      .then(({data: pizzerias}) => {
+        // ...
+      });
+
+Showing records on the map is done like this. (Note: it would even be simpler if Leaflet `wouldn't have followed <https://macwright.org/2016/07/15/longitude-latitude-is-the-right-way.html>`_ GMaps and used {long, lat} instead of {lat, long}.)
```

Comment:
This is a third conditional, where the `if` part should be in the past and the `then` part should have `would`. So `if Leaflet hadn't followed`. (I think `it would be even simpler` is good, although the grammar advice I see online seems to think it should be `it would have been even simpler`.)